diff --git a/app/build.gradle b/app/build.gradle
index f0995b6..05b95d4 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -4,7 +4,7 @@ android {
     compileSdkVersion 27
     defaultConfig {
         applicationId "com.example.azatsepin.theguardianreader"
-        minSdkVersion 21
+        minSdkVersion 23
         targetSdkVersion 27
         versionCode 1
         versionName "0.1"
@@ -24,9 +24,6 @@ android {
             minifyEnabled false
             proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
         }
-        debug {
-
-        }
     }
 }

diff --git a/app/src/main/AndroidManifest.xml b/app/src/main/AndroidManifest.xml
index acc4c18..8d72afd 100644
--- a/app/src/main/AndroidManifest.xml
+++ b/app/src/main/AndroidManifest.xml
@@ -22,7 +22,14 @@
             android:parentActivityName=".MainActivity"
             android:theme="@style/AppTheme.NoActionBar"/>

-        <service android:name=".UpdateService" />
+        <service android:name=".UpdateService"
+            android:permission="android.permission.BIND_JOB_SERVICE" />
+
+        <receiver android:name="StartServiceReceiver" >
+            <intent-filter>
+                <action android:name="android.intent.action.BOOT_COMPLETED" />
+            </intent-filter>
+        </receiver>
     </application>

 </manifest>
\ No newline at end of file
diff --git a/app/src/main/java/com/example/azatsepin/theguardianreader/DetailsActivity.java b/app/src/main/java/com/example/azatsepin/theguardianreader/DetailsActivity.java
index c125302..c73469b 100644
--- a/app/src/main/java/com/example/azatsepin/theguardianreader/DetailsActivity.java
+++ b/app/src/main/java/com/example/azatsepin/theguardianreader/DetailsActivity.java
@@ -7,9 +7,11 @@ import android.support.annotation.Nullable;
 import android.support.design.widget.FloatingActionButton;
 import android.support.v7.app.AppCompatActivity;
 import android.support.v7.widget.Toolbar;
+import android.text.Html;
 import android.view.View;
 import android.widget.Button;
 import android.widget.ImageView;
+import android.widget.TextView;

 import com.airbnb.lottie.LottieAnimationView;
 import com.example.azatsepin.theguardianreader.datasource.AsyncArticleRepository;
@@ -32,12 +34,19 @@ public class DetailsActivity extends AppCompatActivity {
         Toolbar toolbar = findViewById(R.id.toolbar);
         setSupportActionBar(toolbar);
         ImageView imageView = findViewById(R.id.header_image);
+        TextView pillar = findViewById(R.id.pillar);
+        TextView section = findViewById(R.id.section);
+        TextView date = findViewById(R.id.date);
+        TextView body = findViewById(R.id.body);
+        TextView weblink = findViewById(R.id.weblink);

         if (getSupportActionBar() != null) {
             getSupportActionBar().setDisplayHomeAsUpEnabled(true);
             getSupportActionBar().setDisplayShowHomeEnabled(true);
         }

+        toolbar.setNavigationOnClickListener(view -> onBackPressed());
+
         progressBar.setVisibility(View.VISIBLE);

         AsyncArticleRepository repository = ((ReaderApp) getApplication()).getRepository();
@@ -68,6 +77,11 @@ public class DetailsActivity extends AppCompatActivity {
                 }
             });

+            pillar.setText(a.getPillarName());
+            section.setText(a.getSectionName());
+            date.setText(a.getWebPublicationDate());
+            body.setText(Html.fromHtml(a.getFields().getBody()));
+            weblink.setText(a.getWebUrl());
             progressBar.setVisibility(View.INVISIBLE);

         });
diff --git a/app/src/main/java/com/example/azatsepin/theguardianreader/MainActivity.java b/app/src/main/java/com/example/azatsepin/theguardianreader/MainActivity.java
index c12f2ff..ccadbbb 100644
--- a/app/src/main/java/com/example/azatsepin/theguardianreader/MainActivity.java
+++ b/app/src/main/java/com/example/azatsepin/theguardianreader/MainActivity.java
@@ -51,5 +51,7 @@ public class MainActivity extends AppCompatActivity {

             @Override public void onPageScrollStateChanged(int state) { }
         });
+
+        ReaderApp.scheduleJob(getApplicationContext());
     }
 }
diff --git a/app/src/main/java/com/example/azatsepin/theguardianreader/ReaderApp.java b/app/src/main/java/com/example/azatsepin/theguardianreader/ReaderApp.java
index 8c2d1c5..33331dd 100644
--- a/app/src/main/java/com/example/azatsepin/theguardianreader/ReaderApp.java
+++ b/app/src/main/java/com/example/azatsepin/theguardianreader/ReaderApp.java
@@ -1,13 +1,18 @@
 package com.example.azatsepin.theguardianreader;

 import android.app.Application;
+import android.app.job.JobInfo;
+import android.app.job.JobScheduler;
 import android.arch.persistence.room.Room;
-import android.content.Intent;
+import android.content.ComponentName;
+import android.content.Context;

 import com.example.azatsepin.theguardianreader.datasource.AsyncArticleRepository;
 import com.example.azatsepin.theguardianreader.datasource.api.GuardianApi;
 import com.example.azatsepin.theguardianreader.datasource.AppDatabase;

+import java.util.Objects;
+
 import retrofit2.Retrofit;
 import retrofit2.converter.gson.GsonConverterFactory;

@@ -32,9 +37,7 @@ public class ReaderApp extends Application {
                 AppDatabase.class)
                 .build();
         repository = new AsyncArticleRepository(db.articleDao());
-        app = this;
-
-        startService(new Intent(getApplicationContext(), UpdateService.class));
+        app = ReaderApp.this;
     }

     public AsyncArticleRepository getRepository() {
@@ -48,4 +51,14 @@ public class ReaderApp extends Application {
     public static ReaderApp getInstance() {
         return app;
     }
+
+    public static void scheduleJob(Context context) {
+        ComponentName serviceComponent = new ComponentName(context, UpdateService.class);
+        JobInfo.Builder builder = new JobInfo.Builder(0, serviceComponent);
+        builder.setMinimumLatency(30 * 1000); // wait at least
+        builder.setOverrideDeadline(3 * 1000); // maximum delay
+        builder.setRequiredNetworkType(JobInfo.NETWORK_TYPE_UNMETERED);
+        JobScheduler jobScheduler = context.getSystemService(JobScheduler.class);
+        Objects.requireNonNull(jobScheduler).schedule(builder.build());
+    }
 }
diff --git a/app/src/main/java/com/example/azatsepin/theguardianreader/StartServiceReceiver.java b/app/src/main/java/com/example/azatsepin/theguardianreader/StartServiceReceiver.java
new file mode 100644
index 0000000..72765f5
--- /dev/null
+++ b/app/src/main/java/com/example/azatsepin/theguardianreader/StartServiceReceiver.java
@@ -0,0 +1,12 @@
+package com.example.azatsepin.theguardianreader;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+
+public class StartServiceReceiver extends BroadcastReceiver {
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        ReaderApp.scheduleJob(context);
+    }
+}
diff --git a/app/src/main/java/com/example/azatsepin/theguardianreader/UpdateService.java b/app/src/main/java/com/example/azatsepin/theguardianreader/UpdateService.java
index 72599e8..adb4339 100644
--- a/app/src/main/java/com/example/azatsepin/theguardianreader/UpdateService.java
+++ b/app/src/main/java/com/example/azatsepin/theguardianreader/UpdateService.java
@@ -1,33 +1,23 @@
 package com.example.azatsepin.theguardianreader;

-import android.app.Service;
-import android.content.Intent;
-import android.os.IBinder;
-import android.support.annotation.Nullable;
+import android.app.job.JobParameters;
+import android.app.job.JobService;
+import android.util.Log;

-import java.util.concurrent.TimeUnit;
+import static android.support.constraint.Constraints.TAG;

-import androidx.work.Constraints;
-import androidx.work.NetworkType;
-import androidx.work.PeriodicWorkRequest;
-import androidx.work.WorkManager;
+public class UpdateService extends JobService {

-public class UpdateService extends Service {
-    @Nullable
     @Override
-    public IBinder onBind(Intent intent) {
-        return null;
+    public boolean onStartJob(JobParameters params) {
+        Log.i(TAG, "onStartJob: ");
+        ReaderApp.scheduleJob(getApplicationContext());
+        return true;
     }

     @Override
-    public int onStartCommand(Intent intent, int flags, int startId) {
-        Constraints constraints = new Constraints.Builder().setRequiredNetworkType
-                (NetworkType.CONNECTED).build();
-        PeriodicWorkRequest workRequest = new PeriodicWorkRequest.Builder(UpdateWorker.class, 15, TimeUnit.MINUTES)
-                .addTag(UpdateWorker.class.getSimpleName())
-                .setConstraints(constraints)
-                .build();
-        WorkManager.getInstance().enqueue(workRequest);
-        return super.onStartCommand(intent, flags, startId);
+    public boolean onStopJob(JobParameters params) {
+        return true;
     }
+
 }
diff --git a/app/src/main/java/com/example/azatsepin/theguardianreader/UpdateWorker.java b/app/src/main/java/com/example/azatsepin/theguardianreader/UpdateWorker.java
deleted file mode 100644
index 28b2ff5..0000000
--- a/app/src/main/java/com/example/azatsepin/theguardianreader/UpdateWorker.java
+++ /dev/null
@@ -1,22 +0,0 @@
-package com.example.azatsepin.theguardianreader;
-
-import android.content.Context;
-import android.support.annotation.NonNull;
-import android.util.Log;
-
-import androidx.work.Worker;
-import androidx.work.WorkerParameters;
-
-public class UpdateWorker extends Worker {
-
-    public UpdateWorker(@NonNull Context context, @NonNull WorkerParameters workerParams) {
-        super(context, workerParams);
-    }
-
-    @NonNull
-    @Override
-    public Result doWork() {
-        Log.i(UpdateWorker.class.getSimpleName(), "doWork: called");
-        return Result.SUCCESS;
-    }
-}
diff --git a/app/src/main/java/com/example/azatsepin/theguardianreader/ui/adapter/ArticleAdapter.java b/app/src/main/java/com/example/azatsepin/theguardianreader/ui/adapter/ArticleAdapter.java
index f853c0f..4815de9 100644
--- a/app/src/main/java/com/example/azatsepin/theguardianreader/ui/adapter/ArticleAdapter.java
+++ b/app/src/main/java/com/example/azatsepin/theguardianreader/ui/adapter/ArticleAdapter.java
@@ -63,8 +63,11 @@ public class ArticleAdapter extends RecyclerView.Adapter<ArticleAdapter.ArticleV
             titleView.setText(article.getWebTitle());
             Picasso.get()
                     .load(article.getFields().getThumbnail())
+                    .fit()
+                    .centerCrop()
                     .into(imageView);
             button.setOnClickListener(v -> ReaderApp.getInstance().getRepository().create(article));
+            button.setChecked(article.isPinned() || article.getId()!=0);
         }
     }
 }
diff --git a/app/src/main/java/com/example/azatsepin/theguardianreader/ui/adapter/ArticlePagedAdapter.java b/app/src/main/java/com/example/azatsepin/theguardianreader/ui/adapter/ArticlePagedAdapter.java
index 908d911..aa4cf3f 100644
--- a/app/src/main/java/com/example/azatsepin/theguardianreader/ui/adapter/ArticlePagedAdapter.java
+++ b/app/src/main/java/com/example/azatsepin/theguardianreader/ui/adapter/ArticlePagedAdapter.java
@@ -57,8 +57,11 @@ public class ArticlePagedAdapter extends PagedListAdapter<Article, ArticlePagedA
             titleView.setText(article.getWebTitle());
             Picasso.get()
                     .load(article.getFields().getThumbnail())
+                    .fit()
+                    .centerCrop()
                     .into(imageView);
             button.setOnClickListener(v -> ReaderApp.getInstance().getRepository().create(article));
+            button.setChecked(article.isPinned() || article.getId()!=0);
         }
     }

diff --git a/app/src/main/res/drawable/dr_dim_effect.xml b/app/src/main/res/drawable/dr_dim_effect.xml
new file mode 100644
index 0000000..aeb83d0
--- /dev/null
+++ b/app/src/main/res/drawable/dr_dim_effect.xml
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android">
+    <gradient
+        android:endColor="#00000000"
+        android:startColor="#80000000"
+        android:angle="270"
+        android:dither="true"
+        />
+</shape>
\ No newline at end of file
diff --git a/app/src/main/res/drawable/ic_arrow.xml b/app/src/main/res/drawable/ic_arrow.xml
new file mode 100644
index 0000000..40164b5
--- /dev/null
+++ b/app/src/main/res/drawable/ic_arrow.xml
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:height="24dp"
+    android:width="24dp"
+    android:viewportWidth="24"
+    android:viewportHeight="24">
+    <path android:fillColor="#000" android:pathData="M8.59,16.58L13.17,12L8.59,7.41L10,6L16,12L10,18L8.59,16.58Z" />
+</vector>
\ No newline at end of file
diff --git a/app/src/main/res/layout/activity_details.xml b/app/src/main/res/layout/activity_details.xml
index aec8321..2557a04 100644
--- a/app/src/main/res/layout/activity_details.xml
+++ b/app/src/main/res/layout/activity_details.xml
@@ -26,7 +26,9 @@
             <FrameLayout
                 android:id="@+id/header_dim"
                 android:layout_width="match_parent"
-                android:layout_height="match_parent">
+                android:layout_height="match_parent"
+                android:foreground="@drawable/dr_dim_effect">
+
                 <ImageView
                     android:id="@+id/header_image"
                     android:layout_width="match_parent"
@@ -37,6 +39,21 @@
                     android:transitionName="@string/activity_transition_animation"
                     tools:src="@drawable/dr_gradient_3_colors"
                     app:layout_collapseMode="parallax"/>
+
+                <ToggleButton
+                    android:id="@+id/action_save"
+                    android:layout_width="20dp"
+                    android:layout_height="24dp"
+                    android:layout_marginBottom="8dp"
+                    android:layout_marginEnd="8dp"
+                    android:background="@drawable/dr_button_book"
+                    android:fitsSystemWindows="true"
+                    android:focusable="false"
+                    android:focusableInTouchMode="false"
+                    android:textOff=""
+                    android:textOn=""
+                    android:layout_gravity="right"
+                    app:layout_collapseMode="parallax" />
             </FrameLayout>

             <ToggleButton
@@ -76,10 +93,8 @@
         android:clipToPadding="false"
         app:layout_behavior="@string/appbar_scrolling_view_behavior">

-        <FrameLayout
-            android:id="@+id/contentFrame"
-            android:layout_width="match_parent"
-            android:layout_height="match_parent"
+        <include
+            layout="@layout/layout_details"
             app:layout_behavior="@string/appbar_scrolling_view_behavior"/>

     </android.support.v4.widget.NestedScrollView>
diff --git a/app/src/main/res/layout/item_article.xml b/app/src/main/res/layout/item_article.xml
index 51495f9..5349fcf 100644
--- a/app/src/main/res/layout/item_article.xml
+++ b/app/src/main/res/layout/item_article.xml
@@ -16,13 +16,29 @@
         android:layout_width="match_parent"
         android:layout_height="wrap_content">

+        <ToggleButton
+            android:id="@+id/action_save"
+            android:layout_width="24dp"
+            android:layout_height="24dp"
+            android:layout_marginEnd="16dp"
+            android:layout_marginTop="16dp"
+            android:background="@drawable/dr_button_book"
+            android:fitsSystemWindows="true"
+            android:focusable="false"
+            android:focusableInTouchMode="false"
+            android:textColor="#FFFFFFFF"
+            android:textOff=""
+            android:textOn=""
+            app:layout_constraintEnd_toEndOf="@+id/media_image"
+            app:layout_constraintTop_toTopOf="parent" />
+
         <ImageView
             android:id="@+id/media_image"
-            android:layout_width="0dp"
+            android:layout_width="match_parent"
             android:layout_height="0dp"
             android:scaleType="centerCrop"
             android:transitionName="@string/activity_transition_animation"
-            app:layout_constraintDimensionRatio="H,1:1"
+            app:layout_constraintDimensionRatio="1:0.7"
             app:layout_constraintTop_toTopOf="parent"
             app:layout_constraintStart_toStartOf="parent"
             app:layout_constraintEnd_toEndOf="parent" />
@@ -60,30 +76,14 @@
             android:id="@+id/sub_text"
             android:layout_width="0dp"
             android:layout_height="wrap_content"
-            android:layout_marginStart="16dp"
+            android:layout_marginBottom="8dp"
             android:layout_marginEnd="16dp"
-            android:layout_marginBottom="16dp"
+            android:layout_marginStart="16dp"
             android:textAppearance="@style/TextAppearance.AppCompat.Body1"
             android:textColor="#B3FFFFFF"
-            app:layout_constraintStart_toStartOf="parent"
-            app:layout_constraintEnd_toEndOf="parent"
-            app:layout_constraintBottom_toTopOf="@+id/action_save" />
-
-        <ToggleButton
-            android:id="@+id/action_save"
-            android:layout_width="24dp"
-            android:layout_height="24dp"
-            android:layout_marginEnd="8dp"
-            android:layout_marginBottom="8dp"
-            android:textColor="#FFFFFFFF"
-            android:textOn=""
-            android:textOff=""
-            android:focusable="false"
-            android:focusableInTouchMode="false"
-            android:fitsSystemWindows="true"
-            android:background="@drawable/dr_button_book"
             app:layout_constraintBottom_toBottomOf="parent"
-            app:layout_constraintEnd_toEndOf="parent" />
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent" />

     </android.support.constraint.ConstraintLayout>

diff --git a/app/src/main/res/layout/layout_details.xml b/app/src/main/res/layout/layout_details.xml
new file mode 100644
index 0000000..e376256
--- /dev/null
+++ b/app/src/main/res/layout/layout_details.xml
@@ -0,0 +1,75 @@
+<?xml version="1.0" encoding="utf-8"?>
+<android.support.constraint.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    xmlns:tools="http://schemas.android.com/tools">
+
+    <TextView
+        android:id="@+id/pillar"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="8dp"
+        android:layout_marginTop="8dp"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        tools:text="Sport" />
+
+    <TextView
+        android:id="@+id/textView"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="8dp"
+        android:layout_marginTop="8dp"
+        android:drawableStart="@drawable/ic_arrow"
+        app:layout_constraintStart_toEndOf="@+id/pillar"
+        app:layout_constraintTop_toTopOf="parent" />
+
+    <TextView
+        android:id="@+id/section"
+        android:layout_width="wrap_content"
+        android:layout_height="21dp"
+        android:layout_marginStart="8dp"
+        android:layout_marginTop="8dp"
+        app:layout_constraintStart_toEndOf="@+id/textView"
+        app:layout_constraintTop_toTopOf="parent"
+        tools:text="Football" />
+
+    <TextView
+        android:id="@+id/date"
+        android:layout_width="wrap_content"
+        android:layout_height="21dp"
+        android:layout_marginEnd="8dp"
+        android:layout_marginStart="296dp"
+        android:layout_marginTop="8dp"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintHorizontal_bias="1.0"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        tools:text="12.11.2018" />
+
+    <TextView
+        android:id="@+id/body"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginBottom="8dp"
+        android:layout_marginEnd="8dp"
+        android:layout_marginStart="8dp"
+        android:layout_marginTop="8dp"
+        app:layout_constraintBottom_toTopOf="@+id/weblink"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/section" />
+
+    <TextView
+        android:id="@+id/weblink"
+        android:layout_width="wrap_content"
+        android:layout_height="21dp"
+        android:layout_marginBottom="8dp"
+        android:layout_marginEnd="8dp"
+        android:layout_marginStart="8dp"
+        android:text="Open in browser"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent" />
+</android.support.constraint.ConstraintLayout>
\ No newline at end of file